### DISCUSSION TITLE
Packaging `Stdune.Ansi_color` as a separate package

### DISCUSSION THREAD

**Discussion by mbarbin** [ocaml/dune#11107] [Ideas]
I started using this module a while ago and I find it really useful in conjunction to using the package `pp` for building styled error messages in a functional way. I would find it easier to re-use in various contexts and published opam packages if it wasn't part of `stdune` (due to the warning about stdune being unstable). It seems it only depends on foundational packages (`pp`, `ordering`, .. ?).

I'm thinking of names such as `pp_tty`, or `pp_ansi`, add a `type t = Style.t list Pp.t` in it, and good to go!

What do you think? The lift/refactor seems easier that `loc` (#10948) at this time, but I haven't actually tried.

You see, the interface of `Pp` is great, but because the type of `tags` is left for the user to instantiate, there isn't any good default for using `pp` for clis, and user facing error messages. My impression is that `Pp_tty.t` would have the potential to become such standard.

**Comment by rgrinberg** [comment]
Sure, sounds fine to me. You would have to lead the effort though

**Reply by mbarbin** [reply]
For now I ended up packaging some early version of a `pp_tty` library with this idea, in this package https://github.com/mbarbin/pplumbing

I don't exclude the possibility of revisiting further down the road and work on some effort to align with dune libs, once I have had a chance to write a few clis with it. Thanks!
